packages/cli/lib/project-generator.js

```diff
@@ -210,7 +211,7 @@ module.exports = class ProjectGenerator extends BaseGenerator {
   scaffold() {
     if (this.shouldExit()) return false;
 
-    this.destinationRoot(this.projectInfo.outdir);
+    this.destinationRoot(utils.untildify(this.projectInfo.outdir));
```

Comment:
We discussed on Slack that this is non-trivial to test. Here is an idea to try:

1. Create a sandbox directory in user's home. Pick a name that will make it clear that this is a sandbox for LB4 tests, because this directory may stay around even after the test suite finished (especially when it was aborted, e.g. by pressing `Ctrl+C`). Suggested name: `~/.lb4sandbox`. Use `os.homedir()` as suggested by Ryan to find out what's the user home directory (`path.join(os.homedir(), '.lb4sandbox')`).
2. Tell the generator to scaffold the new project in this special sandbox directory, use `~` in the path.
3. Verify that the generator expanded `~` and created the artifacts in the expected place.

Thoughts?

Reply:
I was thinking to test it by:

1. Check if folder exists: `homeDir/somePath/loopback-next/sandbox/my-app` (should be false)
2. For any local test, get pwd: `homeDir/somePath/loopback-next`

3. Input path: `~/somePath/loopback-next/sandbox/my-app`

4. Then check if folder exists: `homeDir/somePath/loopback-next/sandbox/my-app` (should be true) 

It might need some helper functions to get `somePath`

Reply:
Note that `os.homedir()` honours `process.env.HOME`, so you can use it for testing in a sandbox:
```
$ node
> os.homedir()
'/Users/rmg'
> process.env.HOME = '/tmp/foo'
'/tmp/foo'
> os.homedir()
'/tmp/foo'
```

Reply:
I didn't know I could pass an input like that. Thanks!